### PR TITLE
Add dashboard unit tests for Authored Hint and Callout I18n

### PR DESCRIPTION
**Pull request by Hamms**
Including tests for malformed I18n data (completing TODO from https://github.com/code-dot-org/code-dot-org/pull/13592)

### CODE REVIEW

File: dashboard/test/models/blockly_test.rb [hunk 1]
```diff
@@ -141,4 +141,84 @@ def assert_equal_xml(a, b)
...
+          }
+        }
+      }
+    }
+
+    I18n.backend.store_translations test_locale, custom_i18n
+
+    level = Level.create(
+      name: level_name,
+      user_id: 0,
```
Comment (ashercodeorg) [inline dashboard/test/models/blockly_test.rb]: AFAIK, the `user_id` is not required. Can it be omitted? If not, it should be a valid (existing) ID.

Also throughout the PR.

File: dashboard/test/models/blockly_test.rb [hunk 2]
```diff
@@ -141,4 +141,84 @@ def assert_equal_xml(a, b)
...
+          {"hint_markdown": "first english markdown", "hint_id": "first"},
+          {"hint_markdown": "second english markdown", "hint_id": "second"},
+        ]
+      )
+    )
+
+    localized_hints = JSON.parse(level.localized_authored_hints)
+
+    assert_equal localized_hints[0]["hint_markdown"], "first test markdown"
+    assert_equal localized_hints[0]["tts_url"], "https://tts.code.org/sharon22k/180/100/1889ea7b2140fc1aef28a2145df32fbb/test_localize_authored_hints.mp3"
```
Comment (ashercodeorg) [inline dashboard/test/models/blockly_test.rb:178]: Can you explain where these URLs are from and/or how they were generated? Ignore if I simply didn't trace back to find their origin (if they are code generated).

File: dashboard/test/models/blockly_test.rb (hunk 1 above)
Comment (Hamms) [reply dashboard/test/models/blockly_test.rb]: It actually is necessary; we only use this style of translation for `custom?` levels, (ie those which are created by a user in levelbuilder), and we identify those by the presence or absense of the user_id

File: dashboard/test/models/blockly_test.rb (hunk 2 above)
Comment (Hamms) [reply dashboard/test/models/blockly_test.rb:178]: They're generated [here](https://github.com/code-dot-org/code-dot-org/blob/ef8ecf78a204dd2d19a24d9f5a39872ae4a29560/dashboard/app/models/blockly.rb#L334) by the `localized_authored_hints` method, based on the content of the translated markdown

File: dashboard/test/models/blockly_test.rb (hunk 1 above)
Comment (Hamms) [reply dashboard/test/models/blockly_test.rb]: Given that in this context we only care about the presence of the field, do you think it's still important that it be a valid ID?

File: dashboard/test/models/blockly_test.rb (hunk 2 above)
Comment (ashercodeorg) [reply dashboard/test/models/blockly_test.rb:178]: Thanks, didn't trace it back.

File: dashboard/test/models/blockly_test.rb (hunk 1 above)
Comment (ashercodeorg) [reply dashboard/test/models/blockly_test.rb]: Our [code](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/level.rb#L32) suggests it is.

That said, do as you please. :)

**Pull request merged**